src/components/Game/Result/Result.js

```diff
@@ -1,36 +1,118 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { Redirect } from 'react-router-dom';
 import { connect } from 'react-redux';
+import classNames from 'classnames';
 
-const ResultView = ({ words, answer }) => {
-    const isAnswerCorrect = () => words.includes(answer);
-
-    return (
-        <div>
-            {
-                (isAnswerCorrect())
-                    ? 'gratulacje, wygrałeś'
-                    : (
-                        `nie udało się, może następnym razem.
-                        poprawne słowa to:
-                        ${words
-                            .map((word) => word.toUpperCase())
-                            .join(', ')}`
-                    )
-            }
-        </div>
-    );
-};
+import classes from './Result.scss';
+import { submitAnswer } from '../../../store/actions/answer';
+
+class ResultView extends Component {
+    componentWillUnmount() {
+        const { dispatch } = this.props;
+        dispatch(submitAnswer(''));
+    }
+
+    isAnswerCorrect = () => {
+        const { words, answer } = this.props;
+        return words.includes(answer);
+    };
+
+    handlePlayAgainClick = () => {
+        const { history } = this.props;
+        history.replace('/game');
+    };
+
+    render() {
+        const { words, answer } = this.props;
+
+        let redirect = null;
+        if (!answer) {
+            redirect = <Redirect to="/game" />;
+        }
+
+        let other = (
+            <h6 className={classes.SectionHeader}>
+                Słowo nie ma anagramów
+            </h6>
+        );
+        if (words.length > 1 || (words.length === 1 && !this.isAnswerCorrect())) {
+            const otherItems = words
+                .filter((word) => word !== answer)
```

Comment:
lines 40-41 are very cool!
it's good to make such computations in the beginning of `render` function
so the general structure is:
```
const … = … // preparing data
return (
    …
);
…

```
and correct me if I'm wrong but it seems if you move `otherItems` up (perhaps `otherAnswers` as `item` does not convey meaning) then you can simplify the condition in line 39 (`otherAnswers.length > 0`)
one can say the same logic is implemented twice now

Reply:
👍